api/grpcserver/grpc_server.go

```diff
@@ -81,8 +83,45 @@ func (s *GrpcServer) GetUserWithUsername(_ context.Context, username *proto.User
 	u := &proto.User{}
 	fmt.Println(username)
 
-	row := s.db.QueryRow("SELECT user_name, real_name, bio, avatar_url FROM users WHERE user_name = $1", username.UserName)
+	err := s.db.QueryRow("SELECT id, user_name, real_name, bio, avatar_url FROM users WHERE user_name = $1", username.UserName).Scan(&u.Id, &u.UserName, &u.RealName, &u.Bio, &u.AvatarUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	u.Followings, err = s.getRelationsFromUser("SELECT follows.to_id as id, users.user_name FROM follows JOIN users ON follows.to_id=users.id WHERE follows.from_id=$1", u.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	u.Followers, err = s.getRelationsFromUser("SELECT follows.from_id, users.user_name FROM follows JOIN users ON follows.from_id=users.id WHERE follows.to_id=$1", u.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	return u, nil
+}
+
+func (s *GrpcServer) getRelationsFromUser(query string, userID string) ([]*proto.User, error) {
+
+	u := []*proto.User{}
+
+	rows, err := s.db.Query(query, userID)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		user := proto.User{}
+
+		err := rows.Scan(&user.Id, &user.UserName)
```

Comment:
Here we rely on the query that is passed in knowing that id and user name need to be in this order. 
Can we pass this line in from outside?
That would also make adding fields later simpler